### PR TITLE
Splunk less searches simultaneously

### PR DESCRIPTION
**Pull request by craffit**

### CODE REVIEW

File: checks.d/splunk_topology.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def __init__(self, instance, init_config):
         })
 
         self.default_polling_interval_seconds = init_config.get('default_polling_interval_seconds', 15)
-
+        self.default_saved_searches_parallel = init_config.get('default_saved_searches_parallel', 3)
```
Comment (craffit) [inline checks.d/splunk_topology.py]: I'd say this should be a setting of SplunkInstanceConfig and we should make this work for both topology and events

File: checks.d/splunk_topology.py [hunk 2]
```diff
@@ -139,6 +148,8 @@ def _dispatch_saved_search(self, instance_config, saved_search):
         # json output_mode is mandatory for response parsing
         parameters["output_mode"] = "json"
 
+        self.log.info("Dispatching saved search: %s." % saved_search.name)
```
Comment (craffit) [inline checks.d/splunk_topology.py]: This seems a bit verbose

File: checks.d/splunk_topology.py [hunk 3]
```diff
@@ -64,7 +65,12 @@ def should_poll(self, time_seconds):
 
 class SplunkTopology(AgentCheck):
     SERVICE_CHECK_NAME = "splunk.topology_information"
-    BATCH_SIZE = 1000
+
+    @staticmethod
+    def chunks(l, n):
+        """Yield successive n-sized chunks from l."""
+        for i in xrange(0, len(l), n):
+            yield l[i:i + n]
```
Comment (craffit) [inline checks.d/splunk_topology.py]: This is better placed in the utils/splunk.py file i'd say

File: checks.d/splunk_topology.py (hunk 1 above)
Comment (whazenberg) [reply checks.d/splunk_topology.py]: I was under the impression, based on a convo with Kostya, that this was not an issue for events, I haven't looked at it though. I can have a look for events as well and put it in a separate PR.

Comment (whazenberg) [reply checks.d/splunk_topology.py]: Fixed for topology

File: checks.d/splunk_topology.py (hunk 3 above)
Comment (whazenberg) [reply checks.d/splunk_topology.py]: Ack

File: checks.d/splunk_topology.py (hunk 2 above)
Comment (whazenberg) [reply checks.d/splunk_topology.py]: set both messages to debug level, I would too like to have the collector log be clean.

**Pull request merged**